mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -611,13 +636,31 @@ struct ConvertVectorStore final : OpConversionPattern<vector::StoreOp> {
       auto backMask = rewriter.create<arith::ConstantOp>(
           loc, DenseElementsAttr::get(subWidthStoreMaskType, maskValues));
 
-      atomicStore(rewriter, loc, memrefBase, currentDestIndex,
-                  cast<VectorValue>(subWidthStorePart), backMask.getResult());
+      subEmulatedWidthStore(rewriter, loc, memrefBase, currentDestIndex,
+                            cast<VectorValue>(subWidthStorePart),
+                            backMask.getResult());
```

Comment:
I would make this more explicit and skip `subEmulatedWidthStore` altogether:
```suggestion
      useAtomicWrites ?
      atomicStore(rewriter, loc, memrefBase, currentDestIndex,
                            cast<VectorValue>(subWidthStorePart),
                            backMask.getResult())
       : rmwStore(rewriter, loc, memrefBase, currentDestIndex,
                            cast<VectorValue>(subWidthStorePart),
                            backMask.getResult());
```

In particular, I wouldn't worry about the signatures being identical. In fact, no harm if these two diverge.

Separately, the names should make it clear that one hook generates "atomic RMW" and the other "non-atomic RMW" (IIUC, both generate **RMW** sequence).

Reply:
I think you are right! I renamed them to `atomicRMWStore` and `nonAtomicRMWStore` as well.